packages/buidler-core/src/internal/buidler-evm/provider/node.ts

```diff
@@ -336,11 +337,16 @@ export class BuidlerNode {
 
     await this._addTransactionToBlock(block, tx);
 
-    const result = await this._vm.runBlock({
-      block,
-      generate: true,
-      skipBlockValidation: true
-    });
+    let result: RunBlockResult;
+    try {
+      result = await this._vm.runBlock({
+        block,
+        generate: true,
+        skipBlockValidation: true
+      });
+    } catch (error) {
+      throw new TransactionExecutionError(error.message);
```

Comment:
How does this interact with the Solidity tracer? I guess this is for unexpected errors? Can we also pass the original error, so we don't loose info for debugging?

Reply:
The error returned from the runBlock is a plain error, so the change to a different Error class shouldn't have an impact on anything downstream.

I will pass the original error as well then.